core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -75,6 +75,8 @@ import org.apache.spark.util._
  */
 class SparkContext(config: SparkConf) extends Logging with ExecutorAllocationClient {
 
+  var isInited: Boolean = false
```

Comment:
>>> I also don't think that a lack of executor messages indicates a disconnection; it's not possible to distinguish from temporary loss of connectivity this way.

All executors send heartbeat to driver at fixed Rate.Over a period of time, all executors are expire,  I think there is a disconnection. Is there any better way to distinguish from temporary loss of connectivity.
Can we check some times? If all executors still expire, we  indicates a disconnection.